nautobot_ssot/contrib.py

```diff
@@ -257,6 +252,16 @@ class NautobotModel(DiffSyncModel):
 
     _model: Model
 
+    @classmethod
+    def get_queryset(cls):
+        """Get the queryset used to load the models data from Nautobot."""
+        parameter_names = list(cls._identifiers) + list(cls._attributes)
+
+        # Here we identify any foreign keys (i.e. fields with '__' in them) so that we can load them directly in the
+        # first query if this function hasn't been overridden.
+        prefetch_related_parameters = [parameter.split("__")[0] for parameter in parameter_names if "__" in parameter]
+        return cls._model.objects.prefetch_related(*prefetch_related_parameters).all()
```

Comment:
How about we have two methods:
- `_get_queryset` used internally to get the actual queryset, call `get_queryset` and adds the `prefetch_related`
- `get_queryset`, which defaults to `return self._model.objects.all()` but may be overwritten

This way we don't loose the performance benefits when overwriting/pass the burden of implementing that to the user